### PR TITLE
docs: Enclave pool explanation

### LINKED ISSUE #926: docs: add instructions on how to use the idle enclaves pre-provisioned in kubernetes
### Where did you see the issue in our documentation?

https://docs.kurtosis.com/k8s

### Issue description

Right now, a user can spin up a pool of warm/idle enclaves for runs in Kubernetes to speed things up. However, its not clear to me how a user can leverage these enclaves. Do I need to specify the specific enclave that I want my package to run in? How does the name of the enclave get returned to me when I spin them up upon engine-start?

This Github issue tracks the addition of more specific instructions & explanations of how to spin up this pool of enclaves in Kubernetes and how to use them in Kubernetes. I'd also love some docs on how to re-use them

### PR DESCRIPTION
**Pull request by leoporoli**
## Description:
Enclave pool explanation

## Is this change user facing?
YES

## References (if applicable):
This is part of the [enclave pool project](https://www.notion.so/kurtosistech/Enclave-Pool-Project-7273552e53e44b0f931a57568cd170a0?d=2f870450d01d4ad7b8af6d0045602a2a#a20862960dad4868b0b9b6eb320b6178)

Fix #926

### CODE REVIEW

File: docs/docs/guides/running-in-k8s.md [hunk 1]
```diff
@@ -70,3 +70,25 @@ Done! Now you can run any Kurtosis command or package just like if you were doin
 :::tip Kurtosis Kloud Early Access
 To switch back to using Kurtosis locally, simply use: `kurtosis cluster set docker`
 :::
+
+
+V. Activate the enclave pool to accelerate the enclave creation time
```
Comment (leeederek) [inline docs/docs/guides/running-in-k8s.md]:
```suggestion
V. [Optional] Activate the enclave pool to accelerate the enclave creation time
```

File: docs/docs/guides/running-in-k8s.md [hunk 2]
```diff
@@ -70,3 +70,25 @@ Done! Now you can run any Kurtosis command or package just like if you were doin
...
+--------------------------------
+
+This step is optional, but we recommend to do it in order to improve the user experience during the enclave creation.
```
Comment (leeederek) [inline docs/docs/guides/running-in-k8s.md]:
```suggestion
This step is optional, but we recommend taking it as it improves the user experience during the enclave creation, specifically regarding speed.
```

File: docs/docs/guides/running-in-k8s.md [hunk 3]
```diff
@@ -70,3 +70,25 @@ Done! Now you can run any Kurtosis command or package just like if you were doin
...
+
+Creating a new enclave from scratch demands several time-consuming engine's tasks and resources creation that can't be omitted.
```
Comment (leeederek) [inline docs/docs/guides/running-in-k8s.md]:
```suggestion
Creating a new enclave from scratch demands several time-consuming engine tasks and the creation of resources.
```

File: docs/docs/guides/running-in-k8s.md [hunk 4]
```diff
@@ -70,3 +70,25 @@ Done! Now you can run any Kurtosis command or package just like if you were doin
...
+
+Kurtosis includes the enclave pool feature that users can enable for reducing the enclave creation time by creating enclaves before these are requested.
```
Comment (leeederek) [inline docs/docs/guides/running-in-k8s.md]:
```suggestion
The enclave pool feature was introduced to reduce the time it takes for a user to run a Kurtosis package in the cloud by spinning up the enclaves before they are needed.
```

File: docs/docs/guides/running-in-k8s.md [hunk 5]
```diff
@@ -70,3 +70,25 @@ Done! Now you can run any Kurtosis command or package just like if you were doin
...
+
+The enclave pool is a functionality of the Kurtosis engine that automatically creates `idle` enclaves, since the engine is started, that will be available to be used when users need to create a new enclave (e.g: when running the `kurtosis enclave add` command)
```
Comment (leeederek) [inline docs/docs/guides/running-in-k8s.md]:
```suggestion
The enclave pool is a functionality of the Kurtosis engine that automatically creates `idle` enclaves, when the engine is started, that are then used whenever users need to create a new enclave (e.g: when running the `kurtosis enclave add` command).
```

File: docs/docs/guides/running-in-k8s.md [hunk 6]
```diff
@@ -70,3 +70,25 @@ Done! Now you can run any Kurtosis command or package just like if you were doin
...
+
+This mechanism speed-up the enclave creation time by using a running `idle` enclave when a new enclave is requested to the engine
```
Comment (leeederek) [inline docs/docs/guides/running-in-k8s.md]:
```suggestion
This mechanism reduces enclave creation time by using a running `idle` enclave when a new enclave is requested from the engine.
```

**Review by leeederek — APPROVED** [review]

**Comment by leoporoli** [comment]
thanks @leeederek for the review and suggestions.

**Pull request merged**